ocaml/xenopsd/xc/service.ml

```diff
@@ -256,31 +258,56 @@ module DaemonMgmt (D : DAEMONPIDPATH) = struct
 
   let name = D.name
 
-  let pid ~xs domid =
+  (* Look up the commandline args for the process pid; *)
+  (* Check that they include expected args. *)
+  let is_cmdline_valid pid expected_args =
     try
```

Comment:
Would be useful to also check the `exe` symlink. Although it is a bit unlikely that there will be a different process with same cmdline args.

Reply:
Again, would use `~pid` here to avoid confusion.